Sources/ArcGISToolkit/Components/Compass/Compass.swift

```diff
@@ -17,21 +17,24 @@ import SwiftUI
 /// A `Compass` (alias North arrow) shows where north is in a `MapView` or
 /// `SceneView`.
 public struct Compass: View {
-    /// A Boolean value indicating whether  the compass should automatically
-    /// hide/show itself when the heading is `0`.
-    private let autoHide: Bool
-    
     /// The opacity of the compass.
     @State private var opacity: Double = .zero
     
+    /// An action to perform when the compass is tapped.
+    private let action: (() async -> Void)?
```

Comment:
A user will likely want to animate the viewpoint (an asynchronous operation), but they may also want to do other things non-asynchronously. By making the action handler async, you've taken that control out of the users hands. If it were non-async, they could easily perform an async operation by wrapping it in a task (`Task { ... }`), just like this component is doing. But as an async closure, there is no way for them to perform an action inside the closure non-asynchronously. Consider making this non-async, just like a button action handler is.

Reply:
Thanks, I made the change.